operator/main.go

```diff
@@ -403,10 +415,17 @@ func onOperatorStartLeading(ctx context.Context) {
 			nm.Resync(context.Background(), time.Time{})
 		}
 	default:
-		startSynchronizingCiliumNodes(NOPNodeManager)
+		startSynchronizingCiliumNodes(NOPNodeManager, wgOperator)
 		nodeManager = &NOPNodeManager
 	}
 
+	if wgOperator != nil {
+		<-k8sCiliumNodesCacheSynced
+		if err := wgOperator.RestoreFinished(); err != nil {
+			log.WithError(err).Warn("Failed to allocate wireguard IP addrs")
```

Comment:
Ditto.

Reply:
Same reasoning as with the WG agent restore. Just that the worst case is that it will take a bit longer until a new node gets a Wireguard tunnel IP allocated.